Test routing

Phil-Barv opened the pull request:
This PR addresses the issue of testing by creating a robust file structure that allows for the collaborative integration of multiple types of tests into the project. I have provided example dummy files in each of the categories to give a better indication of what goes where. As the branch currently is, the tests for routes.py have been fully implemented.

Phil-Barv committed:
fixed modals forms on Pablo's branch

Phil-Barv committed:
created tests for the backend routes with appropriate documentation

Phil-Barv committed:
updated to include appropriate documentation

Phil-Barv committed:
updated readme to include instructions for running tests

Phil-Barv committed:
removed unused packages from imports

Phil-Barv committed:
added empty files as examples of required tests

cyrushhc commented:
Hi @Phil-Barv, when running the file, I ran into the error
```
ERROR: Could not find a version that satisfies the requirement dataclasses==0.8 (from versions: 0.1, 0.2, 0.3, 0.4, 0.5, 0.6)
ERROR: No matching distribution found for dataclasses==0.8
```

Any idea why this might be the case

Phil-Barv commented:
@cyrushhc We (@pabloguarneros and I) are aware of this problem. It seems its not supported by python3.6. For the time being as we try to come up with a solution just comment it out then pip install the rest.

cyrushhc commented:
I see. I tried doing that and the app.py works. However, after that. I did 
```
cd tests
python3 -m unittest discover
```

but 
1. It says that module "requests" could not be found. So I pip install requests,
2. but then I ran the test again, but got all 9 errors.

Did this happen to you?

cyrushhc commented:
Ok, figured out. We need to run app.py first. I will update the READ.ME

Phil-Barv commented:
@cyrushhc is the app running (localhost 5000)? The tests send requests directly to the flask app.


Update: @cyrushhc  I have updated the readme.MD to explicitly state you need the flask app running. :)

Phil-Barv committed:
updated readme to include that app needs to be running

Corrected an oversight that assumed the flask app would be running to an explicit sentence that says you *must have the flask app running.

Phil-Barv committed:
updated requirements to include requests version 2.27.1

Humancyyborg reviewed (commented):
@Phil-Barv  
-----Good job on the test route implementation.
   --I cant even think of a case that wasn't covered
---I noticed it seems as if you are also testing the auth implementation here, not just routes. 
because I see tests for tokens/not created, logout etc? 
--don't you think we should separate them?

pabloguarneros reviewed (approved):
@Phil-Barv This works and any changes that were resolved are also passing the frontend tests. I resolved the conflicts we previously talked about, but I think it is better for  @Humancyyborg to resolve the last conflict as he is more familiar with the create_project implementation.
